solidity/contracts/integrator/TBTCDepositorProxy.sol

```diff
@@ -0,0 +1,358 @@
+// SPDX-License-Identifier: GPL-3.0-only
+
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+// ██████████████     ▐████▌     ██████████████
+// ██████████████     ▐████▌     ██████████████
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+//               ▐████▌    ▐████▌
+
+pragma solidity 0.8.17;
+
+import {BTCUtils} from "@keep-network/bitcoin-spv-sol/contracts/BTCUtils.sol";
+import {Initializable} from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
```

Comment:
I think we can do it just for this abstract contract. ERC7201 seems to be worth it. Let me see.

Reply:
Unfortunately, this won't be so simple. Migration from OZ v4 to v5 is non-trivial and will incur the upgrade of the OZ Upgrade plugin. This may break things while dealing with existing tBTC contract deployments. I don't want to take this risk right now.

However, to make `TBTCDepositorProxy` usable with both OZ v4 and v5, I decided to remove the dependency on `Initializable` in favor of a custom guard in the `__TBTCDepositorProxy_initialize` initializer. Regarding the storage layout protection, I think we should keep using storage gaps in this case. That gives the best flexibility as child contracts of `TBTCDepositorProxy` can still inherit from OZ v5. This is safe because ERC-7201 namespaces used by OZ v5 are safe against collisions with the standard storage layout used by Solidity (source: [ERC-7201 abstract](https://eips.ethereum.org/EIPS/eip-7201#abstract)). I have also explored the possibility of using ERC-7201 in `TBTCDepositorProxy`. Unfortunately, this is not an option because the OZ Upgrade Plugin raises an error for Solidity versions `<0.8.20` ([source](https://docs.openzeppelin.com/upgrades-plugins/1.x/writing-upgradeable#namespaced-storage-layout)) and `TBTCDepositorProxy` must support all `0.8.x` versions.